src/coord/rmr/rmr.c

```diff
@@ -529,7 +529,15 @@ MRIteratorCtx *MRIteratorCallback_GetCtx(MRIteratorCallbackCtx *ctx) {
 }
 
 void MRIteratorCallback_AddReply(MRIteratorCallbackCtx *ctx, MRReply *rep) {
-  MRChannel_Push(ctx->ic->chan, rep);
+  PushErrorMask mask = MRChannel_Push(ctx->ic->chan, rep);
+  if (mask & CHANNEL_CLOSED) {
+    // We are using a verbose log level to prevent flodding the logs with this message in production
+    RedisModule_Log(RSDummyContext, "verbose", "Tried pushing a reply to a closed channel");
+    MRReply_Free(rep);
+  }
+  if (mask & BROADCAST_FAILURE) {
+    RedisModule_Log(RSDummyContext, "warning", "Failed broadcasting the reply in the channel");
```

Comment:
@kei-nan Since every `MRChannel_Push` may also fail with flag `BROADCAST_FAILURE`, `warning` level could bloat log. Why not use `verbose` level here as well?

Reply:
That is what I tended to do at the start.

Why I chose to use a warning level is because it is more tied to an OS system call failure than a logic flow failure.

Broadcast failure means `pthread_cond_broadcast` failed which is a very fundamental system call which we don't expect to fail very often.
On the other hand if we have a logic flow somewhere which is consistent then the channel being closed can happen quite a bit.

Reply:
So ideally we prefer to keep the `warning` level, but let's reduce the risk of spamming the log so it is  suitable for a patch release.